trio/_version.py

```diff
@@ -1,3 +0,0 @@
-# This file is imported from __init__.py and exec'd from setup.py
-
-__version__ = "0.22.0+dev"
```

Comment:
We want to keep this here because we need to inspect `trio.__version__` at runtime. Hatchling supports this.

Reply:
Can't you use `pkg_resources.get_distribution("trio").version`for that?

Reply:
I've had bad experiences in the past with the overhead of pkg_resources but maybe I didn't use it correctly. It's also semi deprecated in favor of importlib.

Most importantly, this seems like a complex solution for a problem we don't currently have.